torchdynamo/guards.py

```diff
@@ -272,6 +274,14 @@ def NN_MODULE_PARAM_NAMES(self, guard):
         ref = self.arg_ref(guard)
         value = self.get(guard.name)
         keys = {k for k, v in value.named_parameters()}
+        if config.eval_model_named_parameter_checks_enabled and not value.training:
```

Comment:
I'd actually expect this to work in training.  The ID should not change, just the values.